source/Octopus.Tentacle.Client/Scripts/ScriptOrchestratorFactory.cs

```diff
@@ -127,19 +133,11 @@ async Task<CapabilitiesResponseV2> GetCapabilitiesFunc(CancellationToken ct)
 
             logger.Verbose($"Discovered Tentacle capabilities: {string.Join(",", tentacleCapabilities.SupportedCapabilities)}");
 
-            if (tentacleCapabilities.HasScriptServiceV3Alpha())
+            // Check if we support any kubernetes script service.
+            // It's implied (and tested) that GetCapabilities will only return Kubernetes or non-Kubernetes script services, never a mix
+            if (tentacleCapabilities.HasAnyKubernetesScriptService())
```

Comment:
This is a key feature of this PR, if the Capabilities service returns _any_ kubernetes script service, we only look for versions in _that_ hierarchy.

If not, we only look for versions of in the `ScriptService` hierarchy